tests/simulators/carla/test_blueprints.py

```diff
@@ -0,0 +1,226 @@
+import pytest
+
+try:
+    import carla
+except ModuleNotFoundError:
+    pytest.skip("carla package not installed", allow_module_level=True)
+
+from test_actions import getCarlaSimulator
+
+from scenic.simulators.carla.blueprints import (
+    advertisementModels,
+    atmModels,
+    barrelModels,
+    barrierModels,
+    benchModels,
+    bicycleModels,
+    boxModels,
+    busStopModels,
+    carModels,
+    caseModels,
+    chairModels,
+    coneModels,
+    containerModels,
+    creasedboxModels,
+    debrisModels,
+    garbageModels,
+    gnomeModels,
+    ironplateModels,
+    kioskModels,
+    mailboxModels,
+    motorcycleModels,
+    plantpotModels,
+    tableModels,
+    trafficwarningModels,
+    trashModels,
+    truckModels,
+    vendingMachineModels,
+    walkerModels,
+)
+from tests.utils import compileScenic, sampleScene
+
+
+def model_blueprint(simulator, mapPath, town, modelType, modelName):
+    code = f"""
+            param map = r'{mapPath}'
+            param carla_map = '{town}'
+            param time_step = 1.0/10
+
+            model scenic.simulators.carla.model
+            ego = new {modelType} with blueprint '{modelName}'
+            terminate after 1 steps
+        """
+    scenario = compileScenic(code, mode2D=True)
+    scene = sampleScene(scenario)
+    simulation = simulator.simulate(scene)
+    obj = simulation.objects[0]
+    assert obj.blueprint == modelName
+
+
+@pytest.mark.parametrize("modelName", carModels)
```

Comment:
Can we compress these to one parameterized test?

Reply:
Yes, that would be great. If Pytest doesn't allow one parameter to depend on another (which wouldn't surprise me), I think it would be fine to only parametrize over the type of object and have a loop inside the test which iterates over the specific blueprints.

Reply:
Ack! I created a dict from which I loop over to further parameterize the `test_blueprints.py` file!